src/internal/components/pluggableVisualizations/pivotTable/widthItemsHelpers.ts

```diff
@@ -0,0 +1,168 @@
+    isAttributeColumnWidthItem(widthItem)
+        ? isAttributeColumnWidthItemVisible(widthItem, filters)
+        : isMeasureColumnWidthItemVisible(widthItem, filters);
+
+const containsMeasureLocator = (widthItem: IMeasureColumnWidthItem): boolean =>
+    widthItem.measureColumnWidthItem.locators.some(locator => isMeasureLocatorItem(locator));
+
+const widthItemLocatorsHaveProperLength = (
+    widthItem: IMeasureColumnWidthItem,
+    measuresCount: number,
+    columnAttributesCount: number,
+): boolean => {
+    return (
+        widthItem.measureColumnWidthItem.locators.length > 0 &&
+        ((measuresCount > 0 &&
+            containsMeasureLocator(widthItem) &&
+            widthItem.measureColumnWidthItem.locators.length === columnAttributesCount + 1) ||
+            (measuresCount === 0 &&
+                !containsMeasureLocator(widthItem) &&
+                widthItem.measureColumnWidthItem.locators.length === columnAttributesCount))
+    );
+};
+
+// removes attribute widthItems with invalid identifiers
+// removes measure widthItems with invalid identifiers and invalid number of locators
+function adaptWidthItemsToPivotTable(
+    originalColumnWidths: ColumnWidthItem[],
+    measureLocalIdentifiers: string[],
+    rowAttributeLocalIdentifiers: string[],
+    columnAttributeLocalIdentifiers: string[],
+    filters: IBucketFilter[],
+): ColumnWidthItem[] {
+    const attributeLocalIdentifiers = [...rowAttributeLocalIdentifiers, ...columnAttributeLocalIdentifiers];
+
+    return originalColumnWidths.reduce((columnWidths: ColumnWidthItem[], columnWidth: ColumnWidthItem) => {
+        if (isMeasureColumnWidthItem(columnWidth)) {
+            // filter out invalid locators
+            const filteredMeasureColumnWidthItem: IMeasureColumnWidthItem = {
+                measureColumnWidthItem: {
+                    ...columnWidth.measureColumnWidthItem,
+                    locators: columnWidth.measureColumnWidthItem.locators.filter(locator => {
+                        // filter out invalid measure locators
+                        if (isMeasureLocatorItem(locator)) {
+                            return includes(
+                                measureLocalIdentifiers,
+                                locator.measureLocatorItem.measureIdentifier,
+                            );
+                        }
+                        // filter out invalid column attribute locators
+                        return includes(
+                            columnAttributeLocalIdentifiers,
+                            locator.attributeLocatorItem.attributeIdentifier,
+                        );
+                    }),
+                },
+            };
+
+            // check the attribute elements vs filters
+            // and proper locators length
+            // TODO: ONE-4449 - which will create widthItems with empty locators
+            if (
+                isWidthItemVisible(filteredMeasureColumnWidthItem, filters) &&
```

Comment:
isWidthItemVisible indicates that it has something to do with item visibility on in the table. However, from the implementation it check a match with filter. Please consider using better name.

Reply:
Done